src/app/auth/views/Auth/SignupBlog.tsx

```diff
@@ -0,0 +1,131 @@
+import { useAppDispatch } from 'app/store/hooks';
+import { useSignUp } from '../../components/SignUp/useSignUp';
+import { useState } from 'react';
+
+import { AppView, IFormValues } from 'app/core/types';
+import { WarningCircle } from 'phosphor-react';
+import TextInput from 'app/auth/components/TextInput/TextInput';
+import PasswordInput from 'app/auth/components/PasswordInput/PasswordInput';
+import { useForm } from 'react-hook-form';
+import signup from './signup';
+
+const SignupAuth = () => {
+  const [loading, setLoading] = useState(false);
+  const dispatch = useAppDispatch();
+
+  const [error, setError] = useState('');
+
+  // FILTER MESSAGES
+  const { doRegister } = useSignUp('activate');
+
+  const {
+    register,
+    formState: { errors },
+    handleSubmit,
+  } = useForm<IFormValues>({
+    mode: 'onChange',
+    defaultValues: {
+      email: '',
+      password: '',
+    },
+  });
+
+  return (
+    <form onSubmit={handleSubmit((e) => signup(e, dispatch, doRegister, setLoading, true, setError))}>
```

Comment:
Is the only difference from the original SignupAuth the appRedirect parameter?
If so, you can reuse it by passing that parameter as a prop to avoid all that duplicate code.